### PR TITLE
[CDS-57817] : Add support for force deletion of services, environments, environment groups

### PR DESCRIPTION
**Pull request by vivekkarnatiharness**
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`

### CODE REVIEW

File: internal/service/platform/environment/resource_environment.go [hunk 1]
```diff
@@ -43,6 +43,12 @@ func ResourceEnvironment() *schema.Resource {
 				Type:        schema.TypeString,
 				Optional:    true,
 			},
+			"forcedelete": {
```
Comment (rathodmeetsatish) [inline internal/service/platform/environment/resource_environment.go:46]: according to naming convention we are following it should be force_delete

File: internal/service/platform/environment_group/resource_environment_group.go [hunk 2]
```diff
@@ -50,6 +50,12 @@ func ResourceEnvironmentGroup() *schema.Resource {
 				Type:        schema.TypeString,
 				Required:    true,
 			},
+			"forcedelete": {
```
Comment (rathodmeetsatish) [inline internal/service/platform/environment_group/resource_environment_group.go:53]: same as above

File: internal/service/platform/service/resource_service.go [hunk 3]
```diff
@@ -29,6 +29,12 @@ func ResourceService() *schema.Resource {
 				Optional:    true,
 				Computed:    true,
 			},
+			"forcedelete": {
+				Description: "Enable this flag for force deletion of services",
+				Type:        schema.TypeString,
```
Comment (rathodmeetsatish) [inline internal/service/platform/service/resource_service.go:34]: same as above

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]
as there is an addition in schema please genrate docs and add a file file in .changelog folder corresponding to your pr number

File: internal/service/platform/environment_group/resource_environment_group.go (hunk 2 above)
Comment (vivekkarnatiharness) [reply internal/service/platform/environment_group/resource_environment_group.go]: done

File: internal/service/platform/service/resource_service.go (hunk 3 above)
Comment (vivekkarnatiharness) [reply internal/service/platform/service/resource_service.go:34]: done

**Review by rathodmeetsatish — APPROVED** [review]

File: helpers/schema.go [hunk 4]
```diff
@@ -162,6 +162,15 @@ func BuildField(d *schema.ResourceData, field string) optional.String {
 	return optional.EmptyString()
 }
 
+func BuildFieldForBoolean(d *schema.ResourceData, field string) optional.Bool {
+	if arr, ok := d.GetOk(field); ok {
+
+		var var1 string = "true"
+		return optional.NewBool(arr.(string) == var1)
```
Comment (yogesh-chauhan) [inline helpers/schema.go:169]: can we not do this directly like 

"true" == arr.(string)

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]

File: helpers/schema.go (hunk 4 above)
Comment (vivekkarnatiharness) [reply helpers/schema.go:169]: No,i am getting syntax error,That untyped variable cant be compared with a typed variable

File: internal/service/platform/environment/resource_environment.go [hunk 5]
```diff
@@ -43,6 +43,12 @@ func ResourceEnvironment() *schema.Resource {
 				Type:        schema.TypeString,
 				Optional:    true,
 			},
+			"force_delete": {
+				Description: "Enable this flag for force deletion of environments",
```
[the remark this answers is not in the document]
Comment (vivekkarnatiharness) [reply internal/service/platform/environment/resource_environment.go:47]: done

**Review by yogesh-chauhan — APPROVED** [review]

**Review by rathodmeetsatish — APPROVED** [review]

**Pull request merged**